diff --git a/easyeditor/models/alphaedit/AlphaEdit_main.py b/easyeditor/models/alphaedit/AlphaEdit_main.py
index 60e28ea..ea57654 100644
--- a/easyeditor/models/alphaedit/AlphaEdit_main.py
+++ b/easyeditor/models/alphaedit/AlphaEdit_main.py
@@ -82,7 +82,11 @@ def apply_AlphaEdit_to_model(
     # If this is the first calculation (i.e., cache_c_new == false), then initialize cache_c first
     if not cache_c_new:
         W_out = nethook.get_parameter(model, f"{hparams.rewrite_module_tmp.format(hparams.layers[-1])}.weight")
-        if "llama" in hparams.model_name.lower() or "qwen" in hparams.model_name.lower():
+        if (
+            "llama" in hparams.model_name.lower()
+            or "qwen" in hparams.model_name.lower()
+            or "gpt-j-6b" in hparams.model_name.lower()
+        ):
             cache_c = torch.zeros((len(hparams.layers), W_out.shape[1], W_out.shape[1]), device="cpu")
         elif "gpt2-xl" in hparams.model_name.lower():
             cache_c = torch.zeros((len(hparams.layers), W_out.shape[0], W_out.shape[0]), device="cpu")